goal_src/levels/rolling/rolling-lightning-mole.gc

```diff
@@ -5,3 +5,1518 @@
+                      (f28-1 (* (- 1.0 f0-10) (-> arg0 flee-info max-boundary-deflection)))
+                      )
+                 (cond
+                   ((< (cos f28-1) f30-0)
+                    (vector--float*! s3-0 s0-0 (-> s2-0 boundary-normal) f30-0)
+                    (let ((s0-1 vector-normalize!))
+                      (set! sv-224 s3-0)
+                      (let ((a1-7 (sin f28-1)))
+                        (s0-1 sv-224 a1-7)
+                        )
+                      )
+                    (let ((s0-2 vector+float*!))
+                      (set! sv-240 s3-0)
+                      (set! sv-256 s3-0)
+                      (let ((s2-1 (-> s2-0 boundary-normal))
+                            (a3-1 (cos f28-1))
+                            )
+                        (s0-2 sv-240 sv-256 s2-1 a3-1)
+                        )
+                      )
+                    )
+                   (else
+                     (let ((v1-15 s3-0))
+                       (set! (-> v1-15 quad) (-> s0-0 quad))
+                       v1-15
+                       )
+                     )
+                   )
+                 )
+               )
+              ((< (cos (-> arg0 flee-info min-reflect-angle)) f30-0)
+               (vector--float*! s3-0 s0-0 (-> s2-0 boundary-normal) f30-0)
+               (vector-normalize! s3-0 (sin (-> arg0 flee-info min-reflect-angle)))
+               (let ((s1-2 vector--float*!)
+                     (s0-4 s3-0)
+                     )
+                 (set! sv-272 s3-0)
+                 (let ((s2-2 (-> s2-0 boundary-normal))
+                       (a3-3 (cos (-> arg0 flee-info min-reflect-angle)))
+                       )
+                   (s1-2 s0-4 sv-272 s2-2 a3-3)
+                   )
+                 )
+               (set! s1-0 #t)
+               )
+              ((< f30-0 (cos (-> arg0 flee-info max-reflect-angle)))
+               (vector--float*! s3-0 s0-0 (-> s2-0 boundary-normal) f30-0)
+               (vector-normalize! s3-0 (sin (-> arg0 flee-info max-reflect-angle)))
+               (let ((s1-4 vector--float*!)
+                     (s0-6 s3-0)
+                     )
+                 (set! sv-288 s3-0)
+                 (let ((s2-3 (-> s2-0 boundary-normal))
+                       (a3-5 (cos (-> arg0 flee-info max-reflect-angle)))
+                       )
+                   (s1-4 s0-6 sv-288 s2-3 a3-5)
+                   )
+                 )
+               (set! s1-0 #t)
+               )
+              (else
+                (vector--float*! s3-0 s0-0 (-> s2-0 boundary-normal) (* 2.0 f30-0))
+                (set! s1-0 #t)
+                )
+              )
+            )
+          )
+        (vector-normalize-copy! arg1 s3-0 (-> arg0 flee-info deflection-max-dist))
+        )
+      )
+    (set! (-> arg0 nav travel quad) (-> gp-0 quad))
+    s1-0
+    )
+  )
+
+(defun fleeing-nav-enemy-adjust-travel ((arg0 fleeing-nav-enemy))
```

Comment:
I think this takes an additional argument that is unused, you could just add an `arg1` `object`.